pkg/cmd/pulumi/new.go

```diff
@@ -390,6 +451,205 @@ func useSpecifiedDir(dir string) (string, error) {
 	return cwd, nil
 }
 
+type AIPromptRequestBody struct {
+	Language       string `json:"language"`
+	Instructions   string `json:"instructions"`
+	Model          string `json:"model"`
+	ResponseMode   string `json:"responseMode"`
+	ConversationID string `json:"conversationId"`
+	ConnectionID   string `json:"connectionId"`
+}
+
+// Iteratively prompt the user for input, sending their input as a prompt tp Pulumi AI
+// Stream the response back to the console, and repeat until the user is done.
+func runAINew(
+	ctx context.Context,
+	args newArgs,
+	opts display.Options,
+	userName string,
+) (conversationURL string, err error) {
+	languageOptions := []string{
+		"TypeScript",
+		"JavaScript",
+		"Python",
+		"Go",
+		"C#",
+	}
+	if args.aiLanguage == "" {
+		if err = survey.AskOne(&survey.Select{
+			Message: "Please select a language for your project:",
+			Options: languageOptions,
+		}, &args.aiLanguage, surveyIcons(opts.Color)); err != nil {
+			return "", err
+		}
+	}
+	var continuePrompt string
+	var connectionID string
+	var conversationID string
+	for continuePrompt, conversationURL, conversationID, connectionID, err = runAINewPromptStep(
+		opts,
+		args.aiLanguage,
+		args.aiPrompt,
+		args.yes,
+		userName,
+		"",
+		"",
+		"",
+		"",
+	); continuePrompt == RefineSelection; continuePrompt,
+		conversationURL,
+		conversationID,
+		connectionID,
+		err = runAINewPromptStep(
+		opts,
+		args.aiLanguage,
+		args.aiPrompt,
+		args.yes,
+		userName,
+		continuePrompt,
+		connectionID,
+		conversationURL,
+		conversationID,
+	) {
+		if err != nil {
+			return "", err
+		}
+	}
+	return conversationURL, err
+}
+
+func sendPromptToPulumiAI(
+	promptMessage string,
+	conversationID string,
+	connectionID string,
+	userName string,
+	language string,
+) (string, string, string, error) {
+	pulumiAIURL := env.AIServiceEndpoint.Value()
+	if pulumiAIURL == "" {
+		pulumiAIURL = "https://www.pulumi.com/ai"
+	}
+	parsedURL, err := url.Parse(pulumiAIURL)
+	if err != nil {
+		return "", "", "", err
+	}
+	requestPath := parsedURL.JoinPath("api", "chat")
+	requestBody := AIPromptRequestBody{
+		Language:       language,
+		Instructions:   promptMessage,
+		Model:          "gpt-4-turbo",
+		ResponseMode:   "code",
+		ConversationID: conversationID,
+		ConnectionID:   connectionID,
+	}
+	marshalledBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return "", "", "", err
+	}
+	userDataCookie := http.Cookie{Name: "pulumi_command_line_user_name", Value: userName}
+	request, err := http.NewRequest("POST", requestPath.String(), bytes.NewReader(marshalledBody))
```

Comment:
Interesting. So you don't actually have to be logged in to Pulumi Cloud to make these requests?

Reply:
You do not - there are some details about auth impl on the AI service that don't require it (specifically, we allow one prompt for unauthenticated users).